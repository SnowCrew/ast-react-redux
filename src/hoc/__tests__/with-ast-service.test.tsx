import { render, screen, within } from "@testing-library/react";
import AstService from "../../services/ast-shop-service";
import WithAstService from "../with-ast-service.hoc";

describe("HOC WithAstService testing", () => {
  const MockWithAstServiceComponent = WithAstService()(() => (
    <div>Test AstService</div>
  ));

  it("renders snapshot", () => {
    const { container } = render(<MockWithAstServiceComponent />);

    expect(container).toMatchSnapshot();
  });

  // it("component have HOC", async () => {
  //   const MockComponent = jest.fn(() => null);
  //   const ComponentWithHoc = WithAstService()(MockComponent);

  //   render(<ComponentWithHoc />);

  //   expect(MockComponent).toBeCalled();
  //   expect(MockComponent).toBeCalledWith(
  //     expect.objectContaining({ AstService: {AstService} }),
  //     expect.anything()
  //   );
  // });
});
