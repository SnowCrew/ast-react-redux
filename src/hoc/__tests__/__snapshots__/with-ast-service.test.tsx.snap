// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HOC WithAstService testing renders snapshot 1`] = `
<div>
  <div>
    Test AstService
  </div>
</div>
`;
